remoteconfig/client.go

```diff
@@ -0,0 +1,223 @@
+package remoteconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"firebase.google.com/go/v4/internal"
+)
+
+const (
+	defaultRemoteConfigEndpoint = "https://firebaseremoteconfig.googleapis.com/v1"
+)
+
+// Client .
+type Client struct {
+	hc        *internal.HTTPClient
+	projectID string
+}
+
+func (c *Client) getRootURL() string {
+	return fmt.Sprintf("%s/projects/%s/remoteConfig", defaultRemoteConfigEndpoint, c.projectID)
+}
+
+// GetRemoteConfig https://firebase.google.com/docs/reference/remote-config/rest/v1/projects/getRemoteConfig
+func (c *Client) GetRemoteConfig(versionNumber string) (*Response, error) {
+	var opts []internal.HTTPOption
+
+	// Optional. Version number of the RemoteConfig to look up.
+	// If not specified, the latest RemoteConfig will be returned.
+	if versionNumber != "" {
+		opts = append(opts, internal.WithQueryParam("versionNumber", versionNumber))
+	}
+
+	var data RemoteConfig
+
+	resp, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodGet,
+			URL:    c.getRootURL(),
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Response{
+		RemoteConfig: data,
+		Etag:         resp.Header.Get("Etag"),
+	}
+
+	return result, nil
+}
+
+// UpdateRemoteConfig https://firebase.google.com/docs/reference/remote-config/rest/v1/projects/updateRemoteConfig
+func (c *Client) UpdateRemoteConfig(eTag string, validateOnly bool) (*Response, error) {
```

Comment:
Same as goes here. In order to stay aligned with the proposed API, this function should be broken down as shown below:
```
func (c *Client) PublishTemplate(ctx context.Context, template *Template) (*Template, error)
func (c *Client) ValidateTemplate(ctx context.Context, template *Template) (*Template, error)
func (c *Client) ForcePublishTemplate(ctx context.Context, template *Template) (*Template, error)
```
Note: they all should return `Template` as well.

If you do not plan to implement all three you can just add `func (c *Client) PublishTemplate(ctx context.Context, template *Template) (*Template, error)` in this PR.

Reply:
Reconsidered and this looks like a good first step. It is fine to leave this as it is. No need to make further changes. Please ignore my comment.